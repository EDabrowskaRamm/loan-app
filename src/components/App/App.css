.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
  text-align: center;
}

.App__header {
  background-color: #282c34;
  font-size: calc(10px + 2vmin);
  color: white;
  padding: 20px;
}

.App__main {
  flex: 1;
  padding: 20px;
}

.App__radio-wrapper {
  margin-bottom: 20px;
}

.App__radio {
  margin-right: 20px;
}

.App__input-wrapper {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  max-width: 500px;
  margin: 0 auto 20px;
}

.App__input {
  display: flex;
  flex-direction: column;
  align-items: baseline;
}

.App__input-box {
  margin-top: 10px;
}

.App__footer {
  background-color: #282c34;
  font-size: calc(8px);
  color: white;
  padding: 20px;
}