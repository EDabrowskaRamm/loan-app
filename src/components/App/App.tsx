import React, { useState, useEffect } from 'react';
import './App.css';

import { ILoanModel } from '../../models/ILoanModel'
import Table from '../Table/Table'
import { getRate, getTableData } from '../../utils/calculations'

type LoanType = "housing" | "car";

const App = () => {
	const [loan, setLoan] = useState<ILoanModel>({
		amount: 0,
		time: 0,
		interest: 3.5
	});
	const [loanType, setLoanType] = useState<LoanType>("housing");
	const [results, setResults] = useState(false);
	const [monthlyRate, setMonthlyRate] = useState("");

	useEffect(() => {
		if (loanType === "housing") {
			setLoan({
				...loan,
				interest: 3.5
			});
		} else {
			setLoan({
				...loan,
				interest: 5
			});
		}
	}, [loanType]);

	const onSelectLoanType = (event: React.ChangeEvent) => {
		const target = event.target as HTMLInputElement;
		setLoanType(target.name as LoanType);
	}

	const onSetAmount = (event: React.ChangeEvent) => {
		const target = event.target as HTMLInputElement;
		setLoan({
			...loan,
			amount: Number(target.value)
		});
	}

	const onSetTime = (event: React.ChangeEvent) => {
		const target = event.target as HTMLInputElement;
		setLoan({
			...loan,
			time: Number(target.value)
		});
	}

	const onGetData = (event: React.MouseEvent) => {
		event.preventDefault();
		const data = getRate(loan);
		setResults(true);
		setMonthlyRate(data);
	}

	const onResetData = () => {
		setLoan({
			amount: 0,
			time: 0,
			interest: 3.5
		});
		setResults(false);
	}

	const getCostOfCredit: number = (loan.time * 12) * Number(monthlyRate);
	const getAmountOfInterest: number = getCostOfCredit - loan.amount;
	const disabledBtn: boolean = loan.amount <= 0 || loan.time <= 0;

	return (
		<div className="App">
			<header className="App__header">
				<h1>simple loan calculator</h1>
			</header>
			<main className="App__main">
				<form className="App__form">
					<div className="App__radio-wrapper">
						<label className="App__radio">
							housing loan
							<input
								type="checkbox"
								checked={loanType === "housing"}
								onChange={onSelectLoanType}
								name="housing"
							/>
						</label>
						<label className="App__radio">
							car loan
							<input
								type="checkbox"
								checked={loanType === "car"}
								onChange={onSelectLoanType}
								name="car"
							/>
						</label>
					</div>
					<div className="App__input-wrapper">
						<label className="App__input">
							desired amount:
							<input
								type="number"
								value={loan.amount}
								onChange={onSetAmount}
								className="App__input-box"
								disabled={results}
							/>
						</label>
						<label className="App__input">
							payback time (in years):
							<input
								type="number"
								value={loan.time}
								onChange={onSetTime}
								className="App__input-box"
								disabled={results}
							/>
						</label>
						<button type="submit" onClick={onGetData} disabled={disabledBtn}>
							calculate
						</button>
						{results && <button type="button" onClick={onResetData}>
							reset
						</button>}
					</div>
				</form>
				{results && <>
					<div>
						<p>
							Total monthly rate: &nbsp;
							<span>{monthlyRate}</span>
						</p>
						<p>
							Full cost of credit: &nbsp;
							<span>{getCostOfCredit.toFixed(2)}</span>
						</p>
						<p>
							Amount of interest: &nbsp;
							<span>{getAmountOfInterest.toFixed(2)}</span>
						</p>
					</div>
					<Table data={getTableData(loan)} />
				</>}
			</main>
			<footer className="App__footer">
				<p>made by EDR</p>
			</footer>
		</div>
	);
}

export default App;
