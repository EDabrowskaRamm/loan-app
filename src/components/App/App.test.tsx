import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

describe('App', () => {

  it('Snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});