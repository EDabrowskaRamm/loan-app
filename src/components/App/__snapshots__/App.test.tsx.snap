// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <h1>
      simple loan calculator
    </h1>
  </header>
  <main
    className="App__main"
  >
    <form
      className="App__form"
    >
      <div
        className="App__radio-wrapper"
      >
        <label
          className="App__radio"
        >
          housing loan
          <input
            checked={true}
            name="housing"
            onChange={[Function]}
            type="checkbox"
          />
        </label>
        <label
          className="App__radio"
        >
          car loan
          <input
            checked={false}
            name="car"
            onChange={[Function]}
            type="checkbox"
          />
        </label>
      </div>
      <div
        className="App__input-wrapper"
      >
        <label
          className="App__input"
        >
          desired amount:
          <input
            className="App__input-box"
            disabled={false}
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
        <label
          className="App__input"
        >
          payback time (in years):
          <input
            className="App__input-box"
            disabled={false}
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
        <button
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          calculate
        </button>
      </div>
    </form>
  </main>
  <footer
    className="App__footer"
  >
    <p>
      made by EDR
    </p>
  </footer>
</div>
`;
